### ISSUE TITLE
Graphical X / Wayland renderer tests on the CI?

### ISSUE THREAD

**Issue by Ambrevar** [atlas-engineer/nyxt#2383]
In https://github.com/atlas-engineer/nyxt/commit/232eddcf26ee311e8b02c8f49de8e27b2bf690d6 I had to disable the renderer tests on the CI because they fail with:

```
<INFO> [14:02:56] Listening to socket #P"/tmp/nyxt-test.socket".
Unable to init server: Could not connect: Connection refused
error: XDG_RUNTIME_DIR not set in the environment.
error: XDG_RUNTIME_DIR not set in the environment
Nested Wayland compositor could not create display socket
```

Even though we are running in headless mode.  I thought https://github.com/atlas-engineer/nyxt/pull/2054 had figured it out, but apparently not.

Any idea?

**Comment by jmercouris** [comment]
You are correct, there is no code starting a window manager. What I do wonder is if the `ubuntu-latest` image includes a display server. It is possible/probable that is why it is failing.

**Comment by jmercouris** [comment]
I should have read more carefully, it includes Wayland! However, maybe something strange happens when there is no GPU/hardware for a display?
